src/main/java/com/ohmyclass/api/components/user/service/crud/impl/UserService.java

```diff
@@ -1,140 +1,137 @@
 package com.ohmyclass.api.components.user.service.crud.impl;
 
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ohmyclass.api.components.role.entity.Role;
 import com.ohmyclass.api.components.user.dto.in.UserChangeInDTO;
 import com.ohmyclass.api.components.user.dto.in.UserInDTO;
 import com.ohmyclass.api.components.user.dto.out.UserOutDTO;
 import com.ohmyclass.api.components.user.entity.User;
 import com.ohmyclass.api.components.user.repository.IUserRepository;
 import com.ohmyclass.api.components.user.service.crud.IUserService;
 import com.ohmyclass.api.components.user.service.mapper.AUserMapper;
-import com.ohmyclass.api.util.communication.CreateResponseService;
+import com.ohmyclass.api.exceptions.ApiRequestException;
 import com.ohmyclass.api.util.communication.Response;
+import com.ohmyclass.security.util.JwtTokenUtil;
 import com.ohmyclass.util.validate.Validate;
 import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Component;
 
-import java.util.Objects;
-import java.util.Optional;
-import java.util.function.Predicate;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.List;
+import java.util.stream.Collectors;
 
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@Log4j2
 @Component
 @AllArgsConstructor
 public class UserService implements IUserService {
 
 	private final IUserRepository userRepo;
+
 	private final AUserMapper userMapper;
 
+	private final PasswordEncoder passwordEncoder;
+
+	private final JwtTokenUtil tokenUtil;
+
 	@Override
-	public Response<UserOutDTO> login(UserInDTO userIn) {
+	public Response<String> register(UserInDTO userIn) {
 
-		Validate.notNull(userIn);
+		Validate.notNull(userIn, "User cannot be null");
 
-		Predicate<Optional<User>> loginIsValid = Optional::isPresent;
+		userRepo.save(userMapper.inDTOToEntity(userIn));
```

Comment:
I guess you wanted to use the now unused private method **saveUser** here? Otherwise, the password is saved without being hashed, afaik.